lib/srv/ctx.go

```diff
@@ -369,6 +369,9 @@ func NewServerContext(ctx context.Context, parent *sshutils.ConnectionContext, s
 		cancel:                 cancel,
 	}
 
+	// Pre-setup log entry in case early failures need to log something.
+	child.initLogEntry()
+
 	authPref, err := srv.GetAccessPoint().GetAuthPreference(ctx)
 	if err != nil {
 		childErr := child.Close()
```

Comment:
`child.Close()` calls `reportStats` which wants to log a message. 

https://github.com/gravitational/teleport/blob/823fc293bff2b2672e5a9b498ee3012e439aec36/lib/srv/ctx.go#L809